Sources/ParseSwift/Objects/ParseObject.swift

```diff
@@ -84,6 +175,28 @@ public extension ParseObject {
     func toPointer() throws -> Pointer<Self> {
         return try Pointer(self)
     }
+
+    func shouldRestoreKey<W>(_ key: KeyPath<Self, W?>,
+                             original: Self) -> Bool where W: Equatable {
+        self[keyPath: key] == nil && original[keyPath: key] != self[keyPath: key]
```

Comment:
@cbaker6 You spoil everyone with these methods.  I vaguely remember someone wanted to set values as `nil`.  This is really elegant.

Reply:
Haha, I try to make it as easy as possible to use. Plus, as you know, leveraging `KeyPath`'s typically makes things a lot easier for the end-user/developer.